sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -329,7 +329,11 @@ abstract class StreamExecution(
           messageParameters = Map(
             "id" -> id.toString,
             "runId" -> runId.toString,
-            "message" -> message))
+            "message" -> message,
+            "queryDebugString" -> toDebugString(includeLogicalPlan = isInitialized),
+            "startOffset" -> committedOffsets.toOffsetSeq(sources, offsetSeqMetadata).toString,
+            "endOffset" -> availableOffsets.toOffsetSeq(sources, offsetSeqMetadata).toString
```

Comment:
When you pass additional parameters, you should add placeholders for them in the error formats in `error-conditions.json` otherwise they are useless. Found them in the PR: https://github.com/apache/spark/pull/48026